### PR TITLE
[iOS] text input methods to only call updateEditState once 

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Make sure iOS text plugin only makes one `updateEditState` call at the end of every `UITextInput` method that changes the state of the text. `setMarkedText: selectedRange:` currently updates twice when it gets called, and the first call reports an inconsistent state, breaking the interaction with the iOS Kana IME.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/59734

## Tests

I added the following tests:

- testUITextInputCallsUpdateEditingStateOnce

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 1]
```diff
@@ -401,11 +400,7 @@ - (UITextRange*)selectedTextRange {
   return [[_selectedTextRange copy] autorelease];
 }
 
-- (void)setSelectedTextRange:(UITextRange*)selectedTextRange {
-  [self setSelectedTextRange:selectedTextRange updateEditingState:YES];
-}
-
-- (void)setSelectedTextRange:(UITextRange*)selectedTextRange updateEditingState:(BOOL)update {
+- (void)_setSelectedTextRange:(UITextRange*)selectedTextRange {
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:403]: In objc we don't prefix method names with `_`

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 2]
```diff
@@ -440,26 +437,29 @@ - (NSString*)textInRange:(UITextRange*)range {
   return [self.text substringWithRange:textRange];
 }
 
-- (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
-  NSRange replaceRange = ((FlutterTextRange*)range).range;
+- (void)_replaceRange:(NSRange)range withText:(NSString*)text {
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:440]: Same here with `_replaceRange:`

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm (hunk 1 above)
Comment (LongCatIsLooong) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Changed to `setSelectedTextRangeLocal`

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m [hunk 3]
```diff
@@ -17,13 +17,27 @@ @interface FlutterTextInputView ()
...
-- (void)testSecureInput {
-  // Setup test.
-  id engine = OCMClassMock([FlutterEngine class]);
-  FlutterTextInputPlugin* textInputPlugin = [[FlutterTextInputPlugin alloc] init];
+@implementation FlutterTextInputPluginTest {
+  id engine;
+  FlutterTextInputPlugin* textInputPlugin;
+}
+
+- (void)setUp {
```
Comment (justinmc) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m:25]: Nice cleanup 👍

**Pull request merged**